DirectProgramming/C++SYCL_FPGA/Tutorials/Features/kernel_args_restrict/src/kernel_args_restrict.cpp

```diff
@@ -16,8 +16,37 @@ using namespace sycl;
 
 // Forward declare the kernel names in the global scope.
 // This FPGA best practice reduces name mangling in the optimization reports.
-class KernelArgsRestrict;
-class KernelArgsNoRestrict;
+class KernelArgsRestrict_Lambda;
+class KernelArgsNoRestrict_Lambda;
+class KernelArgsRestrict_Functor;
+class KernelArgsNoRestrict_Functor;
```

Comment:
```suggestion
class IDArgsRestrict_Lambda;
class IDArgsNoRestrict_Lambda;
class IDArgsRestrict_Functor;
class IDArgsNoRestrict_Functor;
```

don't feel pressure to rename this. I just prefer including `ID` in these class declarations since they exist only to give a unique ID to your kernels.